core/types/calls.cc

```diff
@@ -338,10 +338,17 @@ unique_ptr<Error> missingArg(const GlobalState &gs, Loc argsLoc, Loc receiverLoc
     return nullptr;
 }
 
-int getArity(const GlobalState &gs, MethodRef method) {
+size_t getArity(const GlobalState &gs, MethodRef method) {
     ENFORCE(!method.data(gs)->arguments.empty(), "Every method should have at least a block arg.");
     ENFORCE(method.data(gs)->arguments.back().flags.isBlock, "Last arg should be the block arg.");
 
+    const auto &arguments = method.data(gs)->arguments;
+    auto restArg =
+        absl::c_find_if(arguments, [&](const auto &arg) { return arg.flags.isRepeated && !arg.flags.isKeyword; });
```

Comment:
> Is this code is looking for an repeated non-keyword arg, and then returning a "infinite" (SIZE_MAX) arity if it exists, since you can pass an arbitrary amount of params in to a method that has a rest arg?

yep

> You could replace

done